### PR TITLE
Remove preserve_to

### LINKED ISSUE #431: Ideal parser settings
Parsing behavior in this library was originally designed for a different data model than the one we now have. The expected usage of `body` and `attributes` have been redefined. As such, the behavior of parsers should be reevaluated.

This is not my first attempt to propose changes to the way parsers behave. I believe my previous attempts have been too focused on proposing _changes_, which is difficult to do without lengthy explanations of the current behavior. The following is instead meant to be read as a standalone proposal for how parsers should behave. Please read accordingly.

I suggest the following would be ideal behavior:
1. Parsing should be non-destructive by default.
2. Parsing should be destructive only when the same value is specified for both `parse_from` and `parse_to`.
3. The `parse_from` setting should default to `body`.
4. The `parse_to` setting should default to `attributes`.
5. The `preserve_to` setting should not exist.

## Rationale

1. Parsing should be non-destructive by default.

The primary purpose of parsing operations is to isolate information. Where the extracted information should be placed is a secondary, but obvious, concern. However, what happens to the original value is easily overlooked. As a general principle, destruction of the original value should require intentional configuration.

2. Parsing should be destructive only when the same value is specified for both `parse_from` and `parse_to`.

A somewhat common use case involves _applying_ structure to an unstructured log. For example, when a user extracts information from a line of text, they may wish keep only the key value pairs they have isolated. When this is the case, an intuitive choice is to overwrite the original value. This should be supported but should require explicit configuration.

3. The `parse_from` setting should default to `body`.

The body typically contains an unstructured log. Parsing this value is the most common parsing use case. Therefore, parsers should default to parsing the body.

4. The `parse_to` setting should default to `attributes`.

The result of a parsing operation is a structured object. This is true of all parsers that have a `parse_to` field.

The data model strongly encourages that structured data belong in `attributes`, so this is a natural default value for `parse_to`.

Additionally, the default value for `parse_to` should be different than that of `parse_from`, else we have a situation where parsing operations are destructive by default.

5. The `preserve_to` setting should not exist.

If parsing is non-destructive by default, there is no need to provide a special setting for preservation. In the case where a user wishes to "back up" a value and destroy the original, they can do so by copying or moving the value before parsing.

### PR DESCRIPTION
**Pull request by djaglowski**
Resolves #431.

Removes the `preserve_to` setting from parsers.

Parsers no longer remove the `parse_from` field, unless promoting the value to a top-level field. (eg `timestamp`, `severity`, `scope`, `trace`)

### CODE REVIEW

File: operator/helper/parser.go [hunk 1]
```diff
@@ -82,6 +81,10 @@ func (c ParserConfig) Build(logger *zap.SugaredLogger) (ParserOperator, error) {
 		parserOperator.TraceParser = c.TraceParser
 	}
 
+	if c.ScopeNameParser != nil {
+		parserOperator.ScopeNameParser = c.ScopeNameParser
+	}
```
Comment (tigrannajaryan) [inline operator/helper/parser.go:86]: Unrelated change?

Comment (djaglowski) [reply operator/helper/parser.go:86]: Sort of. A bug I found while updating tests.

I'll split it into a separate PR.

Comment (djaglowski) [reply operator/helper/parser.go:86]: https://github.com/open-telemetry/opentelemetry-log-collection/pull/465

**Review by jsirianni — APPROVED** [review]
